### PR TITLE
Implement RandAugment layer

### PR DESCRIPTION
**Pull request by LukeWood**
magnitude=1.0, layers=3
weak

magnitude=5.0, layers=3
rand_aug_medium

magnitude=10.0, layers=3
strong

### CODE REVIEW

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    Args:
+        num_layers: the number of layers to use in the rand augment policy.
+        magnitude: the magnitude to use for each of the augmentation.
+        magnitude_standard_deviation: the standard deviation to use when drawing values
+            for the perturbations.
+        probability_to_apply:  the probability to apply an augmentation at each layer.
+        value_range: the value range of the incoming images
+
+    Usage:
+    # TODO(lukewood): document fully.
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/rand_augment.py:47]: This will need to be done before I submit

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+def auto_contrast_policy(magnitude, magnitude_std):
+    return {}
+
+
+def equalize_policy(magnitude, magnitude_std):
+    return {}
+
+
+def solarize_policy(magnitude, magnitude_std):
+    # TODO(lukewood): this should support a sample-able factor.
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/rand_augment.py:156]: We can solve the KerasCV layer issues before v1

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        mean=magnitude / 10.0,
+        standard_deviation=magnitude_std / 10.0,
+        min_value=0,
+        max_value=1,
+    )
+    return {"factor": factor}
+
+
+def contrast_policy(magnitude, magnitude_std):
+    # TODO(lukewood): should we integrate RandomContrast with `factor`?
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/rand_augment.py:180]: This is in core keras, unfortunately it can't take `factor` object.  maybe core keras can somehow support this?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    # RandomContrast layer errors when factor=0
+    factor = max(magnitude / 10, 0.001)
+    return {"factor": factor}
+
+
+def brightness_policy(magnitude, magnitude_std):
+    # TODO(lukewood): should we integrate RandomBrightness with `factor`?
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/rand_augment.py:187]: This is in core keras, unfortunately it can't take factor object. maybe core keras can somehow support this?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+        mean=magnitude / 10.0,
+        standard_deviation=magnitude_std / 10.0,
+        min_value=0,
+        max_value=1,
+    )
+    return {"x_factor": 0, "y_factor": factor}
+
+
+def translate_x_policy(magnitude, magnitude_std):
+    # TODO(lukewood): should we integrate RandomTranslation with `factor`?
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/rand_augment.py:212]: This is in core keras, unfortunately it can't take factor object. maybe core keras can somehow support this?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+    return {"width_factor": magnitude / 10, "height_factor": 0}
+
+
+def translate_y_policy(magnitude, magnitude_std):
+    # TODO(lukewood): should we integrate RandomTranslation with `factor`?
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/rand_augment.py:217]: This is in core keras, unfortunately it can't take factor object. maybe core keras can somehow support this?

**Comment by LukeWood** [comment]
@qlzh727 @fchollet This should be ready now.  Let me know if you see any issues with the layer.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 2 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:179]: I opened a github issue, we will just have to remember to do this.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 1 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py]: Done

**Comment by LukeWood** [comment]
ok, @fchollet @qlzh727 this layer should actually be fully correct now.

**Review by qlzh727 — COMMENTED** [review]
Could we repopulate the PR info with latest changes? (eg how the image looks like with light/medium/heavy augmentation)

**Comment by LukeWood** [comment]
> Could we repopulate the PR info with latest changes? (eg how the image looks like with light/medium/heavy augmentation)

@qlzh727 updated - they look a lot more coherent now!  Thanks!

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+    References:
+        - [RandAugment](https://arxiv.org/abs/1909.13719)
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.
+        num_layers: the number of layers to use in the rand augment policy.
+        magnitude: the magnitude to use for each of the augmentation.
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:43]: do we have a range for the magnitude?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+
+    References:
+        - [RandAugment](https://arxiv.org/abs/1909.13719)
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.
+        num_layers: the number of layers to use in the rand augment policy.
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:42]: Same here, is there a range for number of layers?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+        magnitude: the magnitude to use for each of the augmentation.
+        magnitude_standard_deviation: the standard deviation to use when drawing values
+            for the perturbations.
+        probability_to_apply:  the probability to apply an augmentation at each layer.
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:46]: Assume this is [0, 1]?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+    return {}
+
+
+def equalize_policy(magnitude, magnitude_std):
+    return {}
+
+
+def solarize_policy(magnitude, magnitude_std):
+    threshold_factor = core.NormalFactorSampler(
+        mean=(magnitude / 10) * 256,
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:172]: Should this be 255?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+        standard_deviation=(magnitude_std / 10.0) * 256,
+        min_value=0,
+        max_value=255,
+    )
+    return {"threshold_factor": threshold_factor}
+
+
+def solarize_add_policy(magnitude, magnitude_std):
+    addition_factor = core.NormalFactorSampler(
+        mean=(magnitude / 10) * 110,
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:182]: The 110 seems to be a quite unique magic number here. Can we extra this into a constant and have some comments for it?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+        standard_deviation=(magnitude_std / 10.0) * 110,
+        min_value=0,
+        max_value=110,
+    )
+    return {"addition_factor": addition_factor, "threshold_factor": 128}
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:187]: Also, why do we choose 128 here?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+
+
+def brightness_policy(magnitude, magnitude_std):
+    # TODO(lukewood): should we integrate RandomBrightness with `factor`?
+    return {"factor": magnitude / 10.0}
+
+
+def shear_x_policy(magnitude, magnitude_std):
+    factor = core.NormalFactorSampler(
+        mean=magnitude / 10.0,
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:218]: Since there are too many 10.0 here (which I assume is the max for magnitude), lets extract it into a constant.

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+        mean=0.75 * (magnitude / 10.0),
+        standard_deviation=0.75 * (magnitude_std / 10.0),
+        min_value=0,
+        max_value=1,
+    )
+    return {"width_factor": factor, "height_factor": factor}
+
+
+policy_pairs = [
+    ("auto_contrast", auto_contrast_policy),
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:257]: Should this be a dict?

Comment (qlzh727) [reply keras_cv/layers/preprocessing/rand_augment.py:257]: better to make it a frozen dict (just in case it is modified and cause issue down the line.)

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+    factor = core.NormalFactorSampler(
+        mean=0.75 * (magnitude / 10.0),
+        standard_deviation=0.75 * (magnitude_std / 10.0),
+        min_value=0,
+        max_value=1,
+    )
+    return {"width_factor": factor, "height_factor": factor}
+
+
+policy_pairs = [
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:256]: Capitalize this so that we don't mix this local variables.

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+
+    def _augment(self, sample):
+        sample["images"] = preprocessing_utils.transform_value_range(
+            sample["images"], self.value_range, (0, 255)
+        )
+        augmented_sample = sample
+        for _ in range(self.num_layers):
+            selected_op = tf.random.uniform(
+                (), maxval=len(self.augmentation_layers) + 1, dtype=tf.int32
+            )
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:140]: we should use seed whenever tf.random ops appears.

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 17]
```diff
@@ -0,0 +1,277 @@
...
+            branch_fns = []
+            for (i, layer) in enumerate(self.augmentation_layers):
+                branch_fns.append((i, lambda: layer(sample)))
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment.py:143]: shouldn't we use `layer(augmented_sample)` here? otherwise each layer will always start from original inputs.

File: keras_cv/layers/preprocessing/rand_augment_test.py [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+        ("float32", tf.float32),
+        ("int32", tf.int32),
+        ("uint8", tf.uint8),
+    )
+    def test_runs_with_dtype_input(self, dtype):
+        rand_augment = layers.RandAugment(value_range=(0, 255))
+        xs = tf.ones((2, 512, 512, 3), dtype=dtype)
+        ys = rand_augment(xs)
+        self.assertEqual(ys.shape, (2, 512, 512, 3))
+
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/rand_augment_test.py:56]: I think we need to have more strong test coverage, eg assert how many layers were used under the hood.

**Review by qlzh727 — CHANGES_REQUESTED** [review]

File: keras_cv/core/factor_sampler/normal_factor_sampler.py [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+    """
+
+    def __init__(self, mean, standard_deviation, min_value, max_value, seed=None):
+        self.mean = mean
+        self.standard_deviation = standard_deviation
+        self.min_value = min_value
+        self.max_value = max_value
+        self.seed = seed
+
+    def __call__(self, shape=None, dtype=tf.float32):
```
Comment (fchollet) [inline keras_cv/core/factor_sampler/normal_factor_sampler.py:53]: Nit: in signatures, use the string `"float32"`  instead of the symbol `tf.float32`

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 20]
```diff
@@ -0,0 +1,277 @@
...
+
+    For each `layer`, the policy selects a random operation from a list of operations.
+    It then samples a random number and if that number is less than
+    `probability_to_apply` applies it to the given image.
+
+    References:
+        - [RandAugment](https://arxiv.org/abs/1909.13719)
+
+    Args:
+        value_range: the range of values the incoming images will have.
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:38]: Wondering if `input_value_range` would be clearer -- there's a bit of unambiguity on whether this is the input range, desired output range, or both.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 8 above)
Comment (fchollet) [reply keras_cv/layers/preprocessing/rand_augment.py:42]: Is it clearer to talk about "transformations" or "distortions" instead of "layers"?

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 9 above)
Comment (fchollet) [reply keras_cv/layers/preprocessing/rand_augment.py:46]: That's what I would have expected -- a magnitude between 0 and 1 where 0 is "nothing happens" and 1 is "max distortion". However it seems the default is 5 and the example uses 10, which makes it hard for users to figure it what's the acceptable range and what happens different points of the range.

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 21]
```diff
@@ -0,0 +1,277 @@
...
+    x_test = rand_augment(x_test)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        num_layers=3,
+        magnitude=5.0,
+        magnitude_standard_deviation=1.5,
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:63]: We use `stddev` as the abbreviation elsewhere in the API (e.g. in random ops).

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 22]
```diff
@@ -0,0 +1,277 @@
...
+        probability_to_apply=None,
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:64]: `rate` or something to that extent? `sample_rate`?

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 16 above)
Comment (fchollet) [reply keras_cv/layers/preprocessing/rand_augment.py:140]: Should we use stateless ops and a Generator?

(This should likely be handled at the level of the base augmentation layer -- maybe it already is)

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 23]
```diff
@@ -0,0 +1,277 @@
...
+                    < self.probability_to_apply,
+                    lambda: augmented_sample,
+                    lambda: sample,
+                )
+
+        augmented_sample["images"] = preprocessing_utils.transform_value_range(
+            augmented_sample["images"], (0, 255), self.value_range
+        )
+        return augmented_sample
+
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:160]: Note: the layer needs to be made serializable (and we should have serialization tests)

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 24]
```diff
@@ -0,0 +1,277 @@
...
+    def __init__(
+        self,
+        value_range,
+        num_layers=3,
+        magnitude=5.0,
+        magnitude_standard_deviation=1.5,
+        probability_to_apply=None,
+        seed=None,
+        **kwargs,
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:66]: Should consider making the set of applicable transformations configurable?

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 25]
```diff
@@ -0,0 +1,277 @@
...
+        self.shear_y = cv_preprocessing.RandomShear(**policy["shear_y"], seed=seed)
+        self.translate_x = cv_preprocessing.RandomTranslation(
+            **policy["translate_x"], seed=seed
+        )
+        self.translate_y = cv_preprocessing.RandomTranslation(
+            **policy["translate_y"], seed=seed
+        )
+        self.cutout = cv_preprocessing.RandomCutout(**policy["cutout"], seed=seed)
+
+        self.augmentation_layers = [
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:115]: We may want to factor this as a separate overridable property so as to make it configurable via easy subclassing.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 20 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:38]: It is actually both.  Does this change the naming recommendation?

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 10 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:172]: Updated - the implementation I based this on used 256 so this was an artifact.  Thanks

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 7 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:43]: Updated to [0, 1].  This is much more consistent with our API.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 8 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:42]: Updated to distortions.  `layers` is too overloaded in the Keras ecosystems to use it here I think as an additional point.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 9 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:46]: Updated to be [0, 1].  Some of the distortions ignore the value so it's not a true no-op.  That is not entirely possible.  We could update our implementation to use the `blend()` method alongside invert but that would change the operation substantially.  Could be an interesting idea though.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 11 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:182]: Added a comment and a constant value.

Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:182]: We could make this user configurable, but not important for v1 IMO.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 14 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:257]: How do we create a frozendict?  I don't think there is one in the standard lib.

Made it a dict though.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 15 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:256]: Done

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 16 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:140]: Done, used the generator.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 17 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:143]: Uh oh!  this was a mistake when porting from an old version.  Come to think of it, there is a bug in the if rate part too.

Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:143]: Fixed it all I think.

File: keras_cv/layers/preprocessing/rand_augment_test.py (hunk 18 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment_test.py:56]: Any ideas for this?  This is a tricky point.  Perhaps inspect a sample with a specific seed?

File: keras_cv/core/factor_sampler/normal_factor_sampler.py (hunk 19 above)
Comment (LukeWood) [reply keras_cv/core/factor_sampler/normal_factor_sampler.py:53]: Let's do a sweep and update them all.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 21 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:63]: Done.  That is a nice abbreviation, nice.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 22 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:64]: Rate is great.  lets use that

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 24 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:66]: I was thinking about this, I wonder if we should offer a higher level abstraction:

provide us a list of layers, we create a random policy from it.  This is more or less what AutoAugment is.  Seems kinda overkill to make RandAugment too configurable.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 25 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:115]: That is an API that may be nice.  I could see that being useful.  Maybe we should think about offering an AutoAugment class that takes a list of layers and produces a policy from it, then have RandAugment be a special case.

File: keras_cv/layers/preprocessing/rand_augment_test.py (hunk 18 above)
Comment (qlzh727) [reply keras_cv/layers/preprocessing/rand_augment_test.py:58]: I think you could config the layer to have a counter in the test (for each invocation of __call__), and then check the total count matches the expected self.num_layers

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 23 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:171]: Done

**Comment by LukeWood** [comment]
@fchollet @qlzh727 how do we feel about "distortions".  Perhaps "distortions_per_image" is more descriptive?  Currently "distortions" may still be too vague.

File: keras_cv/layers/preprocessing/rand_augment_test.py (hunk 18 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment_test.py:58]: So would we include this counter in the end code?  Would it fire on every invocation?  Aren't tf.Variable increments are locking? 

I'm not convinced we should include logic in the layer specifically for unit testing is a good pattern in terms of maintaining simplicity.  It introduces a new bug surface for us to consider and makes the code less readable.

**Comment by LukeWood** [comment]
I introduced a new helper method, this abstraction greatly simplifies the code.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 24 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:73]: Overkill is the wrong word - but it seems like the "wrong abstraction" for what we want to offer.

We want to offer a pipeline that allows users to randomly apply layers from a selection of layers.  RandAugment happens to be one of those pipelines, that happens to be well researched.  But I don't think the RandAugment class should handle this responsibility

**Comment by LukeWood** [comment]
> I introduced a new helper method, this abstraction greatly simplifies the code.

Nevermind, this actually made unit tests fail due to an autograph edge case.  Weird!

File: keras_cv/layers/preprocessing/rand_augment_test.py (hunk 18 above)
Comment (qlzh727) [reply keras_cv/layers/preprocessing/rand_augment_test.py:58]: oh, I was suggesting for test only, eg monkey patch the call method with some extra logic in the base layer. eg:

```
def call(self, inputs, **kwargs):
   if not hasattr(self, "call_counter"):
      self.call_counter = 0
   self.call_counter += 1
   super().call(inputs, **kwargs)   

BaseImageAugmentationLayer.call = call
```

File: keras_cv/layers/preprocessing/rand_augment.py [hunk 26]
```diff
@@ -0,0 +1,305 @@
...
+    x_test = rand_augment(x_test)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        distortions=3,
+        magnitude=0.5,
+        magnitude_stddev=1.5,
```
Comment (fchollet) [inline keras_cv/layers/preprocessing/rand_augment.py:70]: QQ: a normal distribution with mean 0.5 and stddev 1.5 will lead to sampled values that are often below 0 or above 1. Is this intended?

**Comment by fchollet** [comment]
> @fchollet @qlzh727 how do we feel about "distortions". 

`num_distortions` or `num_augmentations` seems good. A possible source of ambiguity with just `distortions` is that it may be a list of distortions instead of a count.

File: keras_cv/layers/preprocessing/rand_augment.py (hunk 26 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment.py:70]: Ah this was a typo.  Thanks for the catch.

**Comment by LukeWood** [comment]
Let's close this PR and move to https://github.com/keras-team/keras-cv/pull/281

File: keras_cv/layers/preprocessing/rand_augment_test.py (hunk 18 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/rand_augment_test.py:58]: Now covered in https://github.com/keras-team/keras-cv/pull/281
